tail: New keep_file_handle and fstat_interval_nsec options for tail i… (#2180)
fluent/fluent-bit-docs: pipeline/inputs/tail.md

| `tag_regex`           | Set a regular expression to extract fields from the filename. For example: `(?<pod_name>[a-z0-9](?:[-a-z0-9]*[a-z0-9])?(?:\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)_(?<namespace_name>[^_]+)_(?<container_name>.+)-(?<container_id>[a-z0-9]{64})\.log$`.                   | _none_    |
| `static_batch_size`   | Set the maximum number of bytes to process per iteration for the monitored static files (files that already exist upon Fluent Bit start).  | `50M`     |
| `file_cache_advise`   | Set the `posix_fadvise` in `POSIX_FADV_DONTNEED` mode. This reduces the usage of the kernel file cache. This option is ignored if not running on Linux.                                                             | `on`      |

> | `keep_file_handle`    | Keep file handles open while tailing files. If `false`, file handles are closed after reading, which is useful for SMB/Samba storage backends that prevent file deletion when handles are open.  However, if `false`, log rotation does not work because that requires an active handle to determine where the original file was rotated to. Rotation will be detected as a truncation, so some logs might be lost. | `true`    |

Reviewer: ```suggestion
| `keep_file_handle`    | Keep file handles open while tailing files. If `false`, file handles are closed after reading, which is useful for SMB/Samba storage backends that prevent file deletion when handles are open. However, if `false`, log rotation does not work because that requires an active handle to determine where the original file was rotated to. Rotation will be detected as a truncation, so some logs might be lost. | `true`    |
```

> | `keep_file_handle`    | Keep file handles open while tailing files. When set to `false`, file handles are closed after reading, which is useful for SMB/SAMBA storage backends that prevent file deletion when handles are open. **Note:** When set to `false`, log rotation detection does not work as it requires an active handle to determine where the original file was rotated to. Rotation will be detected as a truncation, so some logs might be lost. | `true`    |

Reviewer: ```suggestion
| `keep_file_handle`    | Keep file handles open while tailing files. If `false`, file handles are closed after reading, which is useful for SMB/Samba storage backends that prevent file deletion when handles are open.  However, if `false`, log rotation does not work because that requires an active handle to determine where the original file was rotated to. Rotation will be detected as a truncation, so some logs might be lost. | `true`    |
```
